redundant-bmc/src/services_impl.cpp

```diff
@@ -202,4 +321,14 @@ std::string ServicesImpl::getFWVersion() const
                        digest[2], digest[3]);
 }
 
+bool ServicesImpl::isPoweredOn() const
+{
+    if (poweredOn.has_value())
+    {
+        return poweredOn.value();
+    }
+
+    throw std::runtime_error("Power state not available");
```

Comment:
I always take a 2nd look when I start seeing exceptions get thrown around in paths that aren't really error paths. It puts a burden on callers of the function and starts to get messy if the call path results in multiple exceptions and people have to start doing if/else in the exception handlers. In which case we need to start documenting the exception names in header definitions and such. But with all that said, I'm ok, as long as the exception handling doesn't start to get into the if/else type stuff.

Reply:
Theoretically I could also put an assert here instead, but an exception gives a bit better error that this function is called when the host state service isn't running, which would only happen if this was called on the passive BMC.